src/main/java/io/github/moulberry/notenoughupdates/core/util/StringUtils.java

```diff
@@ -71,7 +72,17 @@ public static String shortNumberFormat(double n) {
 		return shortNumberFormat(n, 0);
 	}
 
-	private static final char[] c = new char[] { 'k', 'm', 'b', 't' };
+	private static final char[] sizeSuffix = new char[]{'k', 'm', 'b', 't'};
+
+	public static String shortNumberFormat(BigInteger bigInteger) {
```

Comment:
Is there a good reason why we use BigInteger? Double should be enough.

Reply:
i dont trust gemstone miners (also doubles suck)

Reply:
dillo miners exist :)

Reply:
to be fair FloatingSand still only has about 1 / 300000 of the 2^52 precision that a double has, but still, there is no need to assume a risk now